### ISSUE TITLE
jcenter() requirement

### ISSUE THREAD

**Issue by realdadfish** [arturbosch/detekt#2571]
When Detekt 1.7.3 is configured using the Gradle plugin, this outputs

```
The project ':project' doesn't have the jcenter() repository in its repositories list. This is required for Detekt to work correctly. Please add jcenter() to the project's repositories { } closure and try again.
```

Now, I'm here in a coorporate environment with restricted internet access. We have a common Artifactory repository where all external repositories, like jcenter, are set up as remote proxy repositories. How can I disable this warning / force Detekt to use whatever repositories have been set up for the project?

**Comment by runningcode** [comment]
Same ^

**Comment by realdadfish** [comment]
I'm also not sure if this is the root of this issue:

```
> Task :module:detektBaseline FAILED
Analyzing 'module/SomeImpl.kt' led to an exception.
The original exception message was: java.lang.ClassCastException: java.lang.String cannot be cast to java.util.List
Running detekt '1.7.2' on Java '1.8.0_231-b11' on OS 'Windows 10'.
If the exception message does not help, please feel free to create an issue on our GitHub page.
```

This is the exact error that I got with 1.7.2 and that was supposed to be fixed with https://github.com/arturbosch/detekt/issues/2561. Now the log says "detekt 1.7.2" is run, I however applied the plugin with version "1.7.3" to my `build.gradle`s buildscript classpath.

**Comment by runningcode** [comment]
As much as the message is intended as maybe a "warning", this message is printed to the console as an "error". In our app with over 100 modules, this error is printed to the console as an error over 100 times. This blocked our upgrade to 1.7.3

**Comment by rock3r** [comment]
@realdadfish no, #2550 has nothing to do with that error, it is neither the cause nor caused by it.

@runningcode would turning it down to a warning solve the issue for you?

**Comment by runningcode** [comment]
By default on our local machines, we print all lifecycle messages and higher which includes warning and errors to the console, so it wouldn't change much. :) Maybe we can add a configuration option to disable this?

**Comment by rock3r** [comment]
Either that, or someone that knows how can improve the heuristics from "is JCenter repo installed" to "did Gradle resolve the dependencies correctly", that would be the actual fix

**Comment by rock3r** [comment]
Possible starting point: https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/ResolvableDependencies.html#getResolutionResult--

**Comment by BraisGabin** [comment]
> I'm also not sure if this is the root of this issue:
> 
> ```
> > Task :module:detektBaseline FAILED
> Analyzing 'module/SomeImpl.kt' led to an exception.
> The original exception message was: java.lang.ClassCastException: java.lang.String cannot be cast to java.util.List
> Running detekt '1.7.2' on Java '1.8.0_231-b11' on OS 'Windows 10'.
> If the exception message does not help, please feel free to create an issue on our GitHub page.
> ```
> 
> This is the exact error that I got with 1.7.2 and that was supposed to be fixed with #2561. Now the log says "detekt 1.7.2" is run, I however applied the plugin with version "1.7.3" to my `build.gradle`s buildscript classpath.

This would be solved, probably with `./gradlew --stop`. I know, kind of hacky. You can open other issue to track this issue.

**Comment by arturbosch** [comment]
> I'm also not sure if this is the root of this issue:
> 
> ```
> > Task :module:detektBaseline FAILED
> Analyzing 'module/SomeImpl.kt' led to an exception.
> The original exception message was: java.lang.ClassCastException: java.lang.String cannot be cast to java.util.List
> Running detekt '1.7.2' on Java '1.8.0_231-b11' on OS 'Windows 10'.
> If the exception message does not help, please feel free to create an issue on our GitHub page.
> ```
> 
> This is the exact error that I got with 1.7.2 and that was supposed to be fixed with #2561. Now the log says "detekt 1.7.2" is run, I however applied the plugin with version "1.7.3" to my `build.gradle`s buildscript classpath.

No, this is my mistake. The Gradle build cache did not recreate the plugin version constant so it stayed at "1.7.2".

If you want to upgrade to 1.7.3 write:
```kt
detekt {
  toolVersion = "1.7.3"
}
```

I will release a bugfix version later today.
